internal/pkg/template/templates/workloads/partials/cf/alb.yml

```diff
@@ -89,7 +90,7 @@ RulePriorityFunctionRole:
                 - elasticloadbalancing:DescribeRules
               Resource: "*"
 
-{{- if .HTTPSListener}}
+{{- if .ALB.IsHTTPS}}
```

Comment:
Are we sure `ALB` is not nil here? We usually only define concrete struct for CFN template to avoid this problem since there's no linter available to know if it is pointer value or concrete value. I would suggest we only use concrete ALB here and use other variable (like `ALB.Enabled` to determine if ALB is enabled or not.

Reply:
So before even calling `alb` cfn template we check for if `.ALB` is not `nil` [here](https://github.com/aws/copilot-cli/pull/4501/files#diff-89efe4a9579f819c8940b629c45537331bac2a4350e29eb00a1b6fac7b076855R141).